frame/dapps-staking/src/lib.rs

```diff
@@ -196,12 +351,39 @@ where
     }
 }
 
+/// Instruction on how to handle reward payout for stakers.
+/// In order to make staking more competitive, majority of stakers will want to
+/// automatically restake anything they earn.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum RewardHandling {
+    /// Rewards are transfered to stakers balance without any further action.
+    OnlyPayout,
+    /// Rewards are transfered to stakers balance and are immediately re-staked
+    /// on the contract from which the reward was received.
+    PayoutAndStake,
+}
+
+impl Default for RewardHandling {
+    fn default() -> Self {
+        RewardHandling::PayoutAndStake
```

Comment:
Shouldn't the default be `OnlyPayout`?

Reply:
We can discuss this.

My motivation is that we should make `PayoutAndStake` default because this is what benefits users more, as well as the ecosystem.

This can be changed in the commit that will add actual support for this.